discovery/dockerswarm/dockerswarm.go

```diff
@@ -0,0 +1,242 @@
+// limitations under the License.
+
+package dockerswarm
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/swarm"
+	"github.com/docker/docker/client"
+	"github.com/go-kit/kit/log"
+	config_util "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+
+	"github.com/prometheus/prometheus/discovery/refresh"
+	"github.com/prometheus/prometheus/discovery/targetgroup"
+	"github.com/prometheus/prometheus/util/strutil"
+)
+
+const (
+	swarmLabel                         = model.MetaLabelPrefix + "dockerswarm_"
+	swarmLabelEndpointPortName         = swarmLabel + "endpoint_port_name"
+	swarmLabelEndpointPortPublishMode  = swarmLabel + "endpoint_port_publish_mode"
+	swarmLabelServicePrefix            = swarmLabel + "service_"
+	swarmLabelServiceID                = swarmLabelServicePrefix + "id"
+	swarmLabelServiceLabelPrefix       = swarmLabelServicePrefix + "label_"
+	swarmLabelServiceName              = swarmLabelServicePrefix + "name"
+	swarmLabelServiceUpdatingStatus    = swarmLabelServicePrefix + "updating_status"
+	swarmLabelTaskPrefix               = swarmLabel + "task_"
+	swarmLabelTaskContainerImage       = swarmLabelTaskPrefix + "container_image"
+	swarmLabelTaskContainerHostname    = swarmLabelTaskPrefix + "container_hostname"
+	swarmLabelTaskContainerLabelPrefix = swarmLabelTaskPrefix + "container_label_"
+	swarmLabelVirtualIPAddrNetmask     = swarmLabel + "virtual_ip_address_netmask"
+	swarmLabelVirtualIPNetworkID       = swarmLabel + "virtual_ip_network_id"
+	swarmLabelNodePrefix               = swarmLabel + "node_"
+	swarmLabelNodeAddress              = swarmLabelNodePrefix + "address"
+	swarmLabelNodeAvailability         = swarmLabelNodePrefix + "availability"
+	swarmLabelNodeEngineVersion        = swarmLabelNodePrefix + "engine_version"
+	swarmLabelNodeHostname             = swarmLabelNodePrefix + "hostname"
+	swarmLabelNodeID                   = swarmLabelNodePrefix + "id"
+	swarmLabelNodeLabelPrefix          = swarmLabelNodePrefix + "label_"
+	swarmLabelNodeManagerAddr          = swarmLabelNodePrefix + "manager_address"
+	swarmLabelNodeManagerLeader        = swarmLabelNodePrefix + "manager_leader"
+	swarmLabelNodeManagerReachability  = swarmLabelNodePrefix + "manager_reachability"
+	swarmLabelNodeName                 = swarmLabelNodePrefix + "name"
+	swarmLabelNodePlatformArchitecture = swarmLabelNodePrefix + "platform_architecture"
+	swarmLabelNodePlatformOS           = swarmLabelNodePrefix + "platform_os"
+	swarmLabelNodeRole                 = swarmLabelNodePrefix + "role"
+	swarmLabelNodeStatus               = swarmLabelNodePrefix + "status"
+)
+
+// DefaultSDConfig is the default Docker Swarm SD configuration.
+var DefaultSDConfig = SDConfig{
+	RefreshInterval: model.Duration(60 * time.Second),
+	Kind:            "services",
+	Port:            80,
+	Scheme:          "https",
```

Comment:
@brian-brazil Here is the code that forces HTTPS as scheme in the client:

https://github.com/moby/moby/blob/a23ca165c919b62da681cf062429ffe36d249a1b/client/client.go#L151

hence scheme is required here

Reply:
I have removed the need to expose those details to end users.